include/savedata.h

```diff
@@ -0,0 +1,114 @@
+#ifndef POKEPLATINUM_SAVEDATA_H
+#define POKEPLATINUM_SAVEDATA_H
+
+#include "constants/savedata/savedata.h"
+#include "constants/savedata/save_table.h"
+
+typedef struct {
```

Comment:
Convention in the project is to specify a name for structs such that a user can choose whether they want to use `struct StructName` or `StructName` as a type-definition. The names can be identical (and, honestly, should be).